tripal/src/Form/TripalImporterForm.php

```diff
@@ -0,0 +1,570 @@
+        '#usage_id' => 0,
+        '#allowed_types' => $class::$file_types,
+        '#cardinality' => $class::$cardinality,
+      ];
+    }
+
+    if (array_key_exists('file_local', $class::$methods) and $class::$methods['file_local'] == TRUE) {
+      $form['file']['file_local'] = [
+        '#title' => t('Server path'),
+        '#type' => 'textfield',
+        '#maxlength' => 5120,
+        '#description' => t('If the file is local to the Tripal server please provide the full path here.'),
+      ];
+    }
+    if (array_key_exists('file_remote', $class::$methods) and $class::$methods['file_remote'] == TRUE) {
+        $form['file']['file_remote'] = [
+          '#title' => t('Remote path'),
+          '#type' => 'textfield',
+          '#maxlength' => 5102,
+          '#description' => t('If the file is available via a remote URL please provide the full URL here.  The file will be downloaded when the importer job is executed.'),
+        ];
+    }
+
+    if ($class::$use_analysis) {
+      // get the list of analyses
+      $sql = "SELECT * FROM {analysis} ORDER BY name";
+      $org_rset = chado_query($sql);
+      $analyses = [];
+      $analyses[''] = '';
+      while ($analysis = $org_rset->fetchObject()) {
+        $analyses[$analysis->analysis_id] = "$analysis->name ($analysis->program $analysis->programversion, $analysis->sourcename)";
+      }
+      $form['analysis_id'] = [
+        '#title' => t('Analysis'),
+        '#type' => t('select'),
+        '#description' => t('Choose the analysis to which the uploaded data will be associated. ' .
+            'Why specify an analysis for a data load?  All data comes from some place, even if ' .
+            'downloaded from a website. By specifying analysis details for all data imports it ' .
+            'provides provenance and helps end user to reproduce the data set if needed. At ' .
+            'a minimum it indicates the source of the data.'),
+        '#required' => $class::$require_analysis,
+        '#options' => $analyses,
+        '#weight' => -14,
+      ];
+    }
+
+    // Retrieve the forms from the custom TripalImporter class
+    // for this loader.
+    $importer = new $class();
+    $element = [];
+    $element_form = $importer->form($element, $form_state);
+    // Quick check to make sure we had an array returned so array_merge() works.
+    if (!is_array($element_form)) {
+      $element_form = arry();
+    }
+
+    // Merge the custom form with our default one.
+    // This way, the custom TripalImporter can use the #weight property
+    // to change the order of their elements in reference to the default ones.
+    $form = array_merge($form, $element_form);
+
+    $form['button'] = [
+      '#type' => 'submit',
+      '#value' => t($class::$button_text),
+      '#weight' => 10,
+    ];
+    return $form;
+  }
+
+  /**
+   * Validate function for the tripal_get_importer_form form().
+   */
+  function tripal_get_importer_form_validate($form, &$form_state) {

```

Comment:
Are these the old D7 versions? If so, they should be removed entirely at this point.